app.js

```diff
@@ -0,0 +1,84 @@
+var path = require('path');
+var fs = require('fs');
+var express = require('express');
+var Router = require('react-router');
+
+var indexStaticWatcher = require('./lib/index-static-watcher').create();
+var PORT = process.env.PORT || 8008;
+var PRODUCTION = (process.env.NODE_ENV === 'production');
+var DIST_DIR = path.join(__dirname, 'dist');
+
+var indexStatic;
+var router;
+var app = express();
+
+var startProdApp = function() {
+  console.log([
+    'Production mode enabled. Note that "npm install" is assumed to',
+    'have recently been run with NODE_ENV="production". If this is not',
+    'the case, some or all static assets may be out of date.'
+  ].join('\n'));
+  indexStaticWatcher.build(function(err, newIndexStatic) {
+    if (err) {
+      throw err;
+    }
+
+    console.log('Built server-side bundle.');
```

Comment:
Hey @toolness, I'm trying to understand the 3 bundled js:

1. `app.bundle.js` (what's this? I can't find out how this is generated from)
2. `commons.bundle.js` (I guess this is from [here](https://github.com/mozilla/teach.webmaker.org/blob/develop/webpack.config.js#L53))?
3. `index-static.bundle.js`

Are the first 2 for browser and `index-static.bundle.js` for server to watch for code changes & generate `/dist`?  Other than being a Webpack novice I think my brain gets confused easily by the types of "server" we have for the site.

So there's a server that generates `/dist` and our static site just serves w/e is in `/dist`?
And there's also a lightweight server that dynamically renders HTML (#585, aka what this PR is for)?
Are the "2" servers 2 different things or 1 server doing two jobs???  :eyes: